internal/api/users/list_test.go

```diff
@@ -0,0 +1,135 @@
+				"course101":                 {CourseID: "course101", CourseName: "Course 101", Role: model.GetCourseUserRole("other")},
+				"course101-with-zero-limit": {CourseID: "course101-with-zero-limit", CourseName: "Course 101 - With Zero Limit", Role: model.GetCourseUserRole("other")},
+			},
+		},
+		{
+			Email: "owner@test.com",
+			Name:  "owner",
+			Role:  model.GetServerUserRole("owner"),
+			Courses: map[string]core.EnrollmentInfo{
+				"course-languages":          {CourseID: "course-languages", CourseName: "Course Using Different Languages.", Role: model.GetCourseUserRole("owner")},
+				"course-with-lms":           {CourseID: "course-with-lms", CourseName: "Course With LMS", Role: model.GetCourseUserRole("owner")},
+				"course-without-source":     {CourseID: "course-without-source", CourseName: "Course Without Source", Role: model.GetCourseUserRole("owner")},
+				"course101":                 {CourseID: "course101", CourseName: "Course 101", Role: model.GetCourseUserRole("owner")},
+				"course101-with-zero-limit": {CourseID: "course101-with-zero-limit", CourseName: "Course 101 - With Zero Limit", Role: model.GetCourseUserRole("owner")},
+			},
+		},
+		{
+			Email: "student@test.com",
+			Name:  "student",
+			Role:  model.GetServerUserRole("user"),
+			Courses: map[string]core.EnrollmentInfo{
+				"course-languages":          {CourseID: "course-languages", CourseName: "Course Using Different Languages.", Role: model.GetCourseUserRole("student")},
+				"course-with-lms":           {CourseID: "course-with-lms", CourseName: "Course With LMS", Role: model.GetCourseUserRole("student")},
+				"course-without-source":     {CourseID: "course-without-source", CourseName: "Course Without Source", Role: model.GetCourseUserRole("student")},
+				"course101":                 {CourseID: "course101", CourseName: "Course 101", Role: model.GetCourseUserRole("student")},
+				"course101-with-zero-limit": {CourseID: "course101-with-zero-limit", CourseName: "Course 101 - With Zero Limit", Role: model.GetCourseUserRole("student")},
+			},
+		},
+		{
+			Email:   "server-admin@test.com",
+			Name:    "server-admin",
+			Role:    model.GetServerUserRole("admin"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+		{
+			Email:   "server-creator@test.com",
+			Name:    "server-creator",
+			Role:    model.GetServerUserRole("creator"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+		{
+			Email:   "server-owner@test.com",
+			Name:    "server-owner",
+			Role:    model.GetServerUserRole("owner"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+		{
+			Email:   "server-user@test.com",
+			Name:    "server-user",
+			Role:    model.GetServerUserRole("user"),
+			Courses: map[string]core.EnrollmentInfo{},
+		},
+	}
+
+	response := core.SendTestAPIRequest(test, core.NewEndpoint(`users/list`), nil)
+
+	if !response.Success {
+		test.Fatalf("Response is not a success: '%v'.", response)
+	}
+
+	responseContent := response.Content.(map[string]any)
```

Comment:
Any reason you are deserializing the response like this instead of the standard way in all the other API tests?
https://github.com/edulinq/autograder-server/blob/main/internal/api/users/get_test.go#L66